### PR TITLE
[FLINK-17289][docs]Translate tutorials/etl.md to Chinese

### PR DESCRIPTION
**Pull request by liying919**
## What is the purpose of the change

Translate tutorials/etl.md to chinese.

## Brief change log

- etl.zh.md is changed into Chinese version.

## Verifying this change

This change is a trivial rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no 
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature?  no

### CODE REVIEW

**Comment by liying919** [comment]
@flinkbot run azure

**Comment by liying919** [comment]
@flinkbot run azure

**Comment by liying919** [comment]
@rmetzger It seems that error occurs while fetching the code. Do you know what should I do

**Comment by liying919** [comment]
@flinkbot run azure

**Comment by rmetzger** [comment]
I manually restarted the failed tasks. It might have been a temporary issue with GitHub.

Since this change is limited to the documentation, the CI build is not very important.

File: docs/training/etl.zh.md [hunk 1]
```diff
@@ -376,76 +320,52 @@ public static class Deduplicator extends RichFlatMapFunction<Event, Event> {
 }
 {% endhighlight %}
 
-When the flatMap method calls `keyHasBeenSeen.value()`, Flink's runtime looks up the value of this
-piece of state _for the key in context_, and only if it is `null` does it go ahead and collect the
-event to the output. It also updates `keyHasBeenSeen` to `true` in this case. 
+当 flatMap 方法调用 `keyHasBeenSeen.value()`，Flink 运行时将在 _当前键的上下文_ 中检索状态的值，只有当它为 `null` 时，才会输出当前事件。这种情况下，它同时也将更新 `keyHasBeenSeen` 为 `true`。
```
Comment (klion26) [inline docs/training/etl.zh.md:323]:
```suggestion
当 flatMap 方法调用 `keyHasBeenSeen.value()` 时，Flink 会在 _当前键的上下文_ 中检索状态值，只有当状态为 `null` 时，才会输出当前事件。这种情况下，它同时也将更新 `keyHasBeenSeen` 为 `true`。
```

File: docs/training/etl.zh.md [hunk 2]
```diff
@@ -131,36 +117,27 @@ public static class NYCEnrichment implements FlatMapFunction<TaxiRide, EnrichedR
 }
 {% endhighlight %}
 
-With the `Collector` provided in this interface, the `flatmap()` method can emit as many stream
-elements as you like, including none at all.
+使用接口中提供的 `Collector` ，`flatmap()` 可以发射你想要的任意数量的元素，也可以一个都不发。
```
[the remark this answers is not in the document]
Comment (liying919) [reply docs/training/etl.zh.md:120]: 这里我改成“发送”吧

File: docs/training/etl.zh.md [hunk 3]
```diff
@@ -103,9 +91,7 @@ public static class Enrichment implements MapFunction<TaxiRide, EnrichedRide> {
 
 ### `flatmap()`
 
-A `MapFunction` is suitable only when performing a one-to-one transformation: for each and every
-stream element coming in, `map()` will emit one transformed element. Otherwise, you will want to use
-`flatmap()`
+`MapFunction` 只适用于一对一的转换：对每个进入算子的流元素，`map()` 将发射一个转换后的元素。对于除此以外的场景，你将要使用 `flatmap()`。
```
[the remark this answers is not in the document]
Comment (liying919) [reply docs/training/etl.zh.md:94]: 这里我改成“仅发送”吧

File: docs/training/etl.zh.md [hunk 4]
```diff
@@ -131,36 +117,27 @@ public static class NYCEnrichment implements FlatMapFunction<TaxiRide, EnrichedR
...
-expensive, since it involves network communication along with serialization and deserialization.
+每个 `keyBy` 会通过 shuffle 来为数据流进行重新分区。总体来说这个开销是很大的，它涉及网络通信、序列化和反序列化。
 
 <img src="{{ site.baseurl }}/fig/keyBy.png" alt="keyBy and network shuffle" class="offset" width="45%" />
 
-In the example above, the key has been specified by a field name, "startCell". This style of key
-selection has the drawback that the compiler is unable to infer the type of the field being used for
-keying, and so Flink will pass around the key values as Tuples, which can be awkward. It is
-better to use a properly typed KeySelector, e.g.,
+在上面的例子中，将 "startCell" 这个字段定义为key。这种选择key的方式有个缺点，就是编译器无法推断用作键的字段的类型，所以 Flink 会将键值作为元组传递，这有时候会比较难处理。所以最好还是使用一个合适的 KeySelector，
```
[the remark this answers is not in the document]
Comment (liying919) [reply docs/training/etl.zh.md:140]: 此处我把key也按规范统一翻译为“键”吧：
```suggestion
在上面的例子中，将 "startCell" 这个字段定义为键。这种选择键的方式有个缺点，就是编译器无法推断用作键的字段的类型，所以 Flink 会将键值作为元组传递，这有时候会比较难处理。所以最好还是使用一个合适的 KeySelector，比如：
```

File: docs/training/etl.zh.md [hunk 5]
```diff
@@ -262,65 +227,51 @@ The output stream now contains a record for each key every time the duration rea
...
-key.
+这是培训中第一个包含状态的流的例子。尽管状态的处理是透明的，Flink必须跟踪每个不同的键的最大时长。
 
-Whenever state gets involved in your application, you should think about how large the state might
-become. Whenever the key space is unbounded, then so is the amount of state Flink will need.
+只要应用中有状态，你就应该考虑状态的大小。如果键值的数量是无限的，那 Flink 的状态需要的空间也同样是无限的。
 
-When working with streams, it generally makes more sense to think in terms of aggregations over
-finite windows, rather than over the entire stream.
+当我们在流上作业时，考虑有限窗口的聚合往往比整个流聚合更有意义。
```
[the remark this answers is not in the document]
Comment (liying919) [reply docs/training/etl.zh.md:236]: 此处改为"在流处理场景中“，这样确实通顺一点

File: docs/training/etl.zh.md [hunk 6]
```diff
@@ -262,65 +227,51 @@ The output stream now contains a record for each key every time the duration rea
...
-implement your own custom aggregations.
+上面用到的 `maxBy()` 只是 Flink 中 `KeyedStream` 上使用的众多聚合函数中的一个。还有一个更通用的 `reduce()` 函数可以用来实现你的自定义聚合。
 
 {% top %}
 
-## Stateful Transformations
+## 有状态的转换
 
-### Why is Flink Involved in Managing State?
+### 为什么 Flink 要参与管理状态？
```
[the remark this answers is not in the document]
Comment (liying919) [reply docs/training/etl.zh.md:246]: “Flink 为什么要参与状态管理？”  是不是通顺一些呢？

File: docs/training/etl.zh.md [hunk 7]
```diff
@@ -231,13 +200,9 @@ DataStream<Tuple2<Integer, Minutes>> minutesByStartCell = enrichedNYCRides
     });
 {% endhighlight %}
 
-Now it is possible to produce a stream that contains only those rides that are the longest rides
-ever seen (to that point) for each `startCell`.
+现在就可以对每个 `startCell` 找到最长的行程，并产生一个流。
```
[the remark this answers is not in the document]
Comment (liying919) [reply docs/training/etl.zh.md:203]: @klion26 这里翻译为 “现在就可以产生一个流，对每个 `startCell`  仅包含那些最长行程的数据。” 你觉得是不是更通顺一些呢

[the remark this answers is not in the document]
Comment (klion26) [reply docs/training/etl.zh.md:203]: 这个翻译我觉得可以的

File: docs/training/etl.zh.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (klion26) [reply docs/training/etl.zh.md:246]: 这个翻译我觉得可以的

**Comment by liying919** [comment]
@klion26 The left comments are resolved. Please help merge it. Thanks a lot :)

**Pull request merged**